### PR TITLE
[KYUUBI #1575] Implement api: /${version}/operations/${operation_identifier}/resultsetmetadata 

### PR DESCRIPTION
**Pull request by simon824**
### _Why are the changes needed?_
#1575
Implement api: /${version}/operations/${operation_identifier}/resultsetmetadata 

- /${version}/operations/${operation_identifier}/resultsetmetadata
  - mapping: ICLIService#getResultSetMetadata
  - desc: get the table schema of the result set via the given operation identifier
  - method: GET
  - params: none
  - returns: an instance of TableSchema

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.readthedocs.io/en/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by simon824** [comment]
cc @yaooqinn @yanghua

File: kyuubi-server/src/test/scala/org/apache/kyuubi/server/api/v1/OperationsResourceSuite.scala [hunk 1]
```diff
@@ -78,7 +78,17 @@ class OperationsResourceSuite extends KyuubiFunSuite with RestFrontendTestHelper
...
+    }
+  }
 
+  test("test get resultSet metadata") {
+    withKyuubiRestServer { (fe, _, _, webTarget: WebTarget) =>
+      val opHandleStr = getOpHandleStr(fe, OperationType.EXECUTE_STATEMENT)
+      val response = webTarget.path(s"api/v1/operations/$opHandleStr/resultsetmetadata")
+        .request(MediaType.APPLICATION_JSON_TYPE).get()
+      assert(200 == response.getStatus)
+      val operationEvent = response.readEntity(classOf[ResultSetMetaData])
```
Comment (yanghua) [inline kyuubi-server/src/test/scala/org/apache/kyuubi/server/api/v1/OperationsResourceSuite.scala:90]: `operationEvent ` -> `metatata`?

File: kyuubi-server/src/main/scala/org/apache/kyuubi/server/api/v1/OperationsResource.scala [hunk 2]
```diff
@@ -81,4 +82,30 @@ private[v1] class OperationsResource extends ApiRequestContext {
           s"for operation handle $operationHandleStr")
     }
   }
+
+  @ApiResponse(
+    responseCode = "200",
+    content = Array(new Content(
+      mediaType = MediaType.APPLICATION_JSON)),
+    description =
+      "get resultSet metadata")
```
Comment (yanghua) [inline kyuubi-server/src/main/scala/org/apache/kyuubi/server/api/v1/OperationsResource.scala:91]: change all the `resultSet ` to be `result set`?

**Review by yanghua — COMMENTED** [review]
@simon824 Left two minor comments

**Review by yanghua — APPROVED** [review]
LGTM

File: kyuubi-server/src/main/scala/org/apache/kyuubi/server/api/v1/dto.scala [hunk 3]
```diff
@@ -69,3 +69,11 @@ case class GetFunctionsRequest(
...
 
 case class OpActionRequest(action: String)
+
+case class ResultSetMetaData(columns: Seq[ColumnDesc])
+
+case class ColumnDesc(
+    columnName: String,
+    comment: String,
+    position: Int,
+    typeDesc: String)
```
Comment (yaooqinn) [inline kyuubi-server/src/main/scala/org/apache/kyuubi/server/api/v1/dto.scala:79]: dataType

Comment (yaooqinn) [reply kyuubi-server/src/main/scala/org/apache/kyuubi/server/api/v1/dto.scala:79]: shall we also keep the precision and scale if any?

File: kyuubi-server/src/main/scala/org/apache/kyuubi/server/api/v1/dto.scala [hunk 4]
```diff
@@ -69,3 +69,13 @@ case class GetFunctionsRequest(
...
+
+case class ResultSetMetaData(columns: Seq[ColumnDesc])
+
+case class ColumnDesc(
+    columnName: String,
+    comment: String,
+    position: Int,
+    dataType: String,
+    precision: Int,
+    scale: Int)
```
Comment (yaooqinn) [inline kyuubi-server/src/main/scala/org/apache/kyuubi/server/api/v1/dto.scala:81]: let's reorder the params to columnName, dataType, columnIndex, precision, scale, comment

Comment (simon824) [reply kyuubi-server/src/main/scala/org/apache/kyuubi/server/api/v1/dto.scala:81]: ok

**Review by yaooqinn — APPROVED** [review]

File: kyuubi-server/src/main/scala/org/apache/kyuubi/server/api/v1/OperationsResource.scala [hunk 5]
```diff
@@ -81,4 +83,42 @@ private[v1] class OperationsResource extends ApiRequestContext {
...
+            tPrimitiveTypeEntry.getType.toString,
+            c.getPosition,
+            precision,
+            scale,
+            c.getComment)
+        }))
+    } catch {
+      case NonFatal(_) =>
+        throw new NotFoundException(
+          s"Error getting result set metadata for operation handle $operationHandleStr")
```
Comment (ulysses-you) [inline kyuubi-server/src/main/scala/org/apache/kyuubi/server/api/v1/OperationsResource.scala:121]: can we also add a unit test for negative case ?

Comment (simon824) [reply kyuubi-server/src/main/scala/org/apache/kyuubi/server/api/v1/OperationsResource.scala:121]: The negative case about `operationHandle` have been tested in other api ut before, the scene is the same.
And if metadata is empty,it won't throw exception.

**Comment by ulysses-you** [comment]
thanks, merging to master

**Pull request closed without merging**